src/fdc3/dotnet/DesktopAgent/src/MorganStanley.ComposeUI.DesktopAgent/Fdc3DesktopAgent.cs

```diff
@@ -1236,197 +1137,109 @@ private string StoreRaisedIntentForTarget(RaiseIntentSpecification raiseIntentSp
-            {
-                return;
+                if (result.AppIntents.TryGetValue(app.Key, out var appIntent))
+                {
+                    appIntent.Apps = appIntent.Apps.Concat(app.Value.Apps);
+                }
+                else
+                {
+                    result.AppIntents.Add(app.Key, app.Value);
+                }
             }
-
-            result.AppIntents = GetAppIntentsFromIntentMetadataCollection(app, null, intentMetadataCollection, result.AppIntents);
         }
-
-        if (targetAppIdentifier != null)
+        catch (Fdc3DesktopAgentException fdc3ex)
         {
-            var fdc3App = apps.FirstOrDefault(x => x.AppId == targetAppIdentifier.AppId);
-            if (fdc3App == default)
+            if (fdc3ex.ErrorType == ResolveError.TargetAppUnavailable)
             {
-                result.Error = ResolveError.TargetAppUnavailable;
-                return result;
+                result.Error = ResolveError.IntentDeliveryFailed;
             }
-
-            FilterAppIntents(fdc3App);
-            return result;
-        }
-
-        foreach (var app in apps)
-        {
-            if (targetAppIdentifier != null && targetAppIdentifier.AppId != app.AppId)
+            else
             {
-                continue;
+                result.Error = fdc3ex.ErrorType;
             }
-
-            FilterAppIntents(app);
         }
 
         return result;
     }
 
     private Dictionary<string, AppIntent> GetAppIntentsFromIntentMetadataCollection(
-        Fdc3App app,
-        string? instanceId,
-        IEnumerable<KeyValuePair<string, IntentMetadata>> intentMetadataCollection,
-        Dictionary<string, AppIntent> appIntents)
+        IEnumerable<FlatAppIntent> intentMetadataCollection)
     {
+        Dictionary<string, AppIntent> appIntents = new Dictionary<string, AppIntent>();
+
         foreach (var intentMetadata in intentMetadataCollection)
         {
-            var appMetadata =
-                new AppMetadata()
-                {
-                    AppId = app.AppId,
-                    InstanceId = instanceId,
-                    Name = app.Name,
-                    Version = app.Version,
-                    Title = app.Title,
-                    Tooltip = app.ToolTip,
-                    Description = app.Description,
-                    Icons = app.Icons == null ? Enumerable.Empty<Icon>() : app.Icons.Select(Icon.GetIcon),
-                    Screenshots = app.Screenshots == null
-                        ? Enumerable.Empty<Screenshot>()
-                        : app.Screenshots.Select(Screenshot.GetScreenshot),
-                    ResultType = intentMetadata.Value.ResultType
-                };
+            var appMetadata = intentMetadata.App.ToAppMetadata(intentMetadata.InstanceId?.ToString(), intentMetadata.Intent.ResultType);
 
-            if (!appIntents.TryGetValue(intentMetadata.Key, out var appIntent)) //Name is null
```

Comment:
Intent Name can be null in the conformance tests, that's why we have used the key of the intentmetadata

Reply:
Now it is ensured to have the Name filled at this point